lib/mozart_fetcher/fetcher.ex

```diff
@@ -11,14 +11,14 @@ defmodule MozartFetcher.Fetcher do
 
   def process(components) do
     ExMetrics.timeframe "function.timing.fetcher.process" do
-      max_timeout =
-        TimeoutParser.max(components) +
-          Application.get_env(:mozart_fetcher, :additional_task_await_timeout)
+      max_timeout = TimeoutParser.max(components)
+      stream_opts = [timeout: max_timeout, on_timeout: :kill_task, max_concurrency: 40]
```

Comment:
`max_concurrency: 40` is obviously a random number. if not passed it defaults to the number of cores.

Reply:
Wondering if in the previous approach we were creating a number of processes equivalent to the size of the `component` list. Maybe for huge pages, the number of concurrent requests would cause the timeout issue? We could probably set max_concurrency with different values and load test this...

Reply:
Sometimes we have 50+ components.

Reply:
Should we add a buffer to the timeout?

Reply:
the size of the batches is hard to define, but could be the root of the previous issue, if we process a page made of say 80 components they will be processed concurrently but given a limited number of cores not in parallel, this could possibly add additional latency to the request? The batches can alleviate this problem as the timeout will be measured against the current batch.
 
ok for the additional buffer! I'd probably add a few milliseconds